src/firefox/index.js

```diff
@@ -189,6 +189,86 @@ export async function run(
 }
 
 
+// isDefaultProfile types and implementation.
+
+const DEFAULT_PROFILES_NAMES = [
+  'default',
+  'dev-edition-default',
+];
+
+export type IsDefaultProfileFn = (
+  profilePathOrName: string,
+  ProfileFinder?: typeof FirefoxProfile.Finder,
+) => Promise<boolean>;
+
+/*
+ * Tests if a profile is a default Firefox profile (both as a profile name or
+ * profile path).
+ *
+ * Returns a promise that resolves to true if the profile is one of default Firefox profile.
+ */
+export async function isDefaultProfile(
+  profilePathOrName: string,
+  ProfileFinder?: typeof FirefoxProfile.Finder = FirefoxProfile.Finder
+): Promise<boolean> {
+  if (DEFAULT_PROFILES_NAMES.indexOf(profilePathOrName) >= 0) {
+    return true;
+  }
+
+  const baseProfileDir = ProfileFinder.locateUserDirectory();
+  const profilesIniPath = path.join(baseProfileDir, 'profiles.ini');
+  try {
+    await fs.stat(profilesIniPath);
+  } catch (error) {
+    if (isErrorWithCode('ENOENT', error)) {
+      // No profiles exist yet, default to false (the default profile name contains a
+      // random generated component).
+      return false;
+    }
+
+    // Re-throw any unexpected exception.
+    throw error;
+  }
+
+  // Check for profile dir path.
+  const finder = new ProfileFinder(baseProfileDir);
+  const readProfiles = promisify(finder.readProfiles, finder);
+
+  await readProfiles();
+
+  const normalizedProfileDirPath = path.normalize(
+    path.join(path.resolve(profilePathOrName), path.sep)
+  );
+
+  for (const profile of finder.profiles) {
+    // Check if the profile dir path or name is one of the default profiles
+    // defined in the profiles.ini file.
+    if (DEFAULT_PROFILES_NAMES.indexOf(profile.Name) >= 0 ||
```

Comment:
@kumar303 the early return should ensure that the usage error with the "Cannot use default profile with --keep-profile-changes..."  message is logged also when the profiles.ini file doesn't exist yet and the user is using "default" or "dev-edition-default" as the firefox profile parameter.